### DISCUSSION TITLE
How to highlight text?

### DISCUSSION THREAD

**Discussion by Jtth3brick** [cymian/mouseless#96] [General]
I can't figure out how to highlight text, although I can't tell if this just isn't a feature or if I'm doing something wrong.

**Comment by cymian** [comment]
hi @Jtth3brick, there is a `hold for drag` action you can assign in the keymap.  I will add a list of actions and explanations to the documentation very soon, but basically how it works is that you hold the key it's assigned to while pressing an 'action' key (space or char) to click, and it will start the drag.  Do it again to keep the drag going, or just do a regular click to release.

if you've been trying that, and it hasn't been working (I find that it generally works to select text and drag windows, but has issues with other types of items), i will track status of the fix for that in #20.

please let me know if you have any more questions!

**Reply by miiraheart** [reply]
could you please make a video on how to do this I still do not get the hang of it, like it works 1/5 times and I still do not know how I did it

**Reply by cymian** [reply]
hi @miiraheart, I will do a clearer tutorial soon!

**Reply by miiraheart** [reply]
Sorry to be impatient but is the tutorial going to be available soon ? Would love to know how to use these actions, for now I am only using clicks and moving mouse and that’s it, I still don’t understand the rest even after reading the available doc. I would love to be able to use all the features and finally stop using my mouse. Thank you in advance.

**Reply by cymian** [reply]
@miiraheart no problem, sorry for the delay.  my original intention was to get another video out sooner, but getting the app to a more polished state first has taken priority.

please let me know if this helps you:

- to right click:
  - make sure the `hold for right button` action is assigned to a keyboard key (it's `CommandRight` by default)
  - bring up the overlay, and press two characters as usual to choose a cell
  - press CommandRight+Space to right click (or if you've assigned `hold for right button` to something else, press that plus space)
    - or press CommandRight+[some character] to right click at the given char in the subgrid
    - make sure you're holding commandRight down when you press space

- to drag:
  - do the same as above, but with CommandLeft instead of CommandRight
  - the overlay should stay up after first action
  - execute another action without CommandLeft to drop/release

- to move:
  - same as above but with OptionLeft

you can also use the `cycle mouse action type` and `cycle mouse button` to set the action/button, so you don't need to hold an extra key down when executing the click.

hope it helps, let me know!

p.s./edit: i'll be adding `hold for actions` or similar for the rest of the mouse buttons soon, so you won't need to use `cycle mouse button` for them

**Reply by miiraheart** [reply]
Thank you for the detailed response. 

I have managed to use mouseless to do all of the actions,  thank you ! 

But would drag work for dragging a file and moving it to a folder ? as for me it only worked with text but not with dragging files.

**Reply by cymian** [reply]
@miiraheart dragging should be fixed in the current [preview release](https://github.com/croian/mouseless-issues/releases/tag/v0.3.0-preview.1) if you want to use it before the official v0.3.0 release :)

**Reply by miiraheart** [reply]
Sorry to bring this up again, but I was wondering how could I copy text and scroll at the same time, I tried the drag option and before hitting the last key to end the drag selection, I tried activating the wheel scroll but it did not work. Is this supported yet ?

**Reply by cymian** [reply]
@miiraheart no problem, thank you for raising that issue.  I'll have to think about how to make that possible.  What comes to mind is allowing `enter wheel mode` while dragging, or while the overlay is up generally, as long as their are no hotkey conflicts.  And/or allowing the upcoming `hold for wheel mode` command while the overlay is up.

**Reply by miiraheart** [reply]
@croian Got it ! For me, this is important as I copy and paste a lot of code daily. I like `enter wheel mode` feature, Looking forward to this !
